### PR TITLE
shard: fix `GetRange` for objects stored in the write-cache

### LINKED ISSUE #1635: Sometimes "object hash --range" and "object range" return response for entire object
## Steps to Reproduce (for bugs)
1. Create container without specifying ACL
2. Put object to the container
3. Put same object to the container with headers
4. Call `object hash --range 0:10` on object from p.3. Sometimes it returns hash of entire object rather than hash of specified range.
5. Call `object range --range 0:10` on object from p.3. Sometimes it returns content of entire object rather than hash of specified range.

Please, find attached [calls of CLI commands](https://github.com/nspcc-dev/neofs-node/files/9209076/cli-log.txt)

## Your Environment
The issue was not observed on devenv, but occurs in cloud deployment of the system

* Version used:
v0.30.0

### PR DESCRIPTION
**Pull request by fyrchik**
Close #1635 

Signed-off-by: Evgenii Stratonikov <evgeniy@morphbits.ru>

### CODE REVIEW

File: pkg/local_object_storage/shard/range.go [hunk 1]
```diff
@@ -102,8 +104,26 @@ func (s *Shard) GetRange(prm RngPrm) (RngRes, error) {
...
+	wc := func(c writecache.Cache) (*object.Object, error) {
+		res, err := c.Get(prm.addr)
+		if err != nil {
+			return nil, err
+		}
+
+		payload := res.Payload()
+		from := rng.GetOffset()
+		to := from + rng.GetLength()
+		if uint64(len(payload)) < to {
```
Comment (carpawell) [inline pkg/local_object_storage/shard/range.go:116]:
```suggestion
		if uint64(len(payload)) < from || uint64(len(payload)) < to {
```
?

Comment (fyrchik) [reply pkg/local_object_storage/shard/range.go]: Fixed

File: pkg/local_object_storage/blobstor/blobovnicza.go [hunk 2]
```diff
@@ -628,7 +628,7 @@ func (b *blobovniczas) getObjectRange(blz *blobovnicza.Blobovnicza, prm GetRange
 	to := from + prm.rng.GetLength()
 	payload := obj.Payload()
 
-	if uint64(len(payload)) < to {
+	if pLen := uint64(len(payload)); to < from || pLen < from || pLen < to {
```
Comment (carpawell) [inline pkg/local_object_storage/blobstor/blobovnicza.go:631]: `to` equals `from` (uint) + `len` (uint) so i doubt it could be less than `from`

**Review by carpawell — APPROVED** [review]

File: pkg/local_object_storage/blobstor/blobovnicza.go (hunk 2 above)
Comment (fyrchik) [reply pkg/local_object_storage/blobstor/blobovnicza.go:631]: Try `uint64(10) + math.MaxUint64`

Comment (carpawell) [reply pkg/local_object_storage/blobstor/blobovnicza.go:631]: oh, overflow...

**Pull request merged**